sootup.java.bytecode/src/main/java/sootup/java/bytecode/frontend/AsmUtil.java

```diff
@@ -154,60 +154,96 @@ public static Type toJimpleSignature(@Nonnull String desc) {
 
   @Nonnull
   public static Type toJimpleType(@Nonnull String desc) {
-    int idx = desc.lastIndexOf('[');
-    int nrDims = idx + 1;
+    int nrDims = countArrayDim(desc);
     if (nrDims > 0) {
-      if (desc.charAt(0) != '[') {
+      desc = desc.substring(nrDims);
+    }
+
+    Type baseType = toPrimitiveOrVoidType(desc).orElse(null);
+    if (baseType == null && desc.charAt(0) != 'L') {
+      throw new AssertionError("Unknown descriptor: " + desc);
+    }
+
+    if (baseType == null && desc.charAt(0) == 'L') {
+      if (desc.charAt(desc.length() - 1) != ';') {
+        throw new AssertionError("Invalid reference descriptor: " + desc);
+      }
+      String name = desc.substring(1, desc.length() - 1);
+      name = toQualifiedName(name);
+      baseType = JavaIdentifierFactory.getInstance().getType(toQualifiedName(name));
+    }
+    if (!(baseType instanceof JavaClassType) && desc.length() > 1) {
```

Comment:
the errormsg does not match instanceof type?

Reply:
no it matches. If basetype is not of instance JavaClassType, it is a primitive datatype. 
But all primitive datatyped are described with one Letter.
This Check is to prevent that "Cool" is not translated to Char primitive because the first Letter is C